homeassistant/components/point/config_flow.py

```diff
@@ -44,109 +45,106 @@ class PointFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow."""
 
     VERSION = 1
+    code: str | None = None
+
+    @property
+    def schema(self):
+        """Return current schema."""
+        return vol.Schema(
+            {
+                vol.Required(CONF_REDIRECT_URI): str,
```

Comment:
Wait, why doesn't this implement an OAuth flow?

Reply:
Yes but not using home-assistant's oauth.

I've been trying to implement using the guidelines but I don't quite understand all parts and have not found an integration that does it by the book either.

The userbase for Point is quite small, perhaps it is better to pull this integration and use hacs instead.

Reply:
> I've been trying to implement using the guidelines but I don't quite understand all parts and have not found an integration that does it by the book either.

There are many integrations doing oauth flows? They all use the same constructs.

The thing is, the redirect URL should not even be a question... With the default flows, it will handle that for the user. 

>  perhaps it is better to pull this integration and use hacs instead.

I do not understand what the latter reasoning has to do with anything? Even as a HACS integration, one should be using the right constructs, right?

Reply:
> There are many integrations doing oauth flows? They all use the same constructs.

Could you please name a few for reference.

> > perhaps it is better to pull this integration and use hacs instead.
> 
> I do not understand what the latter reasoning has to do with anything? Even as a HACS integration, one should be using the right constructs, right?

The reasoning is that I don't feel I have the time to keep up with the high code quality standards. I did another PR https://github.com/home-assistant/core/pull/100843 but that was rejected due to 1. I updated the yaml 2. Once I've rewritten the flow to not use yaml the pr should be split up, this is the first stage of the split.

I feel that I've already spent too much time on this, implementing "the right" oauth flow will take me another 6-8 hours which I don't have atm. Either accept this pr (with guidelines on how to make it appropriate) or pull support for Point (the integration is currently broken as Minut have change the oauth flow).

Reply:
> Could you please name a few for reference.

Search for `AbstractOAuth2FlowHandler` usage in config_flow.py files across our codebase, we currently have 26 integrations implementing it.

Some ones that have been implemented fairly recently (and thus had recent reviews) are Twitch, Fitbit and YouTube.

> The reasoning is that I don't feel I have the time to keep up with the high code quality standards.

This has nothing to do with coding standards. 🤷 

> Either accept this pr (with guidelines on how to make it appropriate) or pull support for Point (the integration is currently broken as Minut have change the oauth flow).

I really want to help, but let me be very clear: These kinds of statements don't help and are not a factor in merging PRs.

Reply:
Looking upstream at the minute point docs, they implement a really standard OAuth2 flow, which we fully support out of the box. The amount of code used by this integration can be reduced quite a bit and simplified.

Should not take more than 1-2 hours to fix/adjust I think (including import of existing configuration).

Reply:
> Looking upstream at the minute point docs, they implement a really standard OAuth2 flow, which we fully support out of the box. The amount of code used by this integration can be reduced quite a bit and simplified.
> 
> Should not take more than 1-2 hours to fix/adjust I think (including import of existing configuration).

Sounds great, I'll have a look at it!